### PR TITLE
webhook properties section

### LINKED ISSUE #376: Specify observe property protocol bindings for HTTP Webhook Profile
Currently the HTTP Webhook Profile specifies protocol bindings for subscribing and unsubscribing from events, but not yet for observing and unobserving properties. It says in the introduction that it covers both, so protocol bindings need to be defined for the following operations:

- `observeproperty`
- `unobserveproperty`
- `observeallproperties`
- `unobserveallproperties`

### PR DESCRIPTION
**Pull request by mlagally**
resolves #376

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-profile/pull/382.html" title="Last updated on Apr 19, 2023, 11:31 AM UTC (09ce906)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-profile/382/27d0390...09ce906.html" title="Last updated on Apr 19, 2023, 11:31 AM UTC (09ce906)">Diff</a>

### CODE REVIEW

**Comment by mlagally** [comment]
Ready for review.

File: index.html [hunk 1]
```diff
@@ -2968,12 +2969,305 @@ <h2>Protocol Binding</h2>
...
+          </ul>
+          <pre class="example">
+              HTTP/1.1 200 OK
+              Content-Type: application/json
+              {
+                  subscriptionId: 1234-4544-1211
+              }
+          </pre>
+          <p><span class="rfc2119-assertion" id="http-webhook-profile-protocol-binding-observeproperty-6">
+              Whenever an event of the specified type occurs, the Web Thing MUST
```
Comment (egekorkan) [inline index.html:3077]: what is meant by event of the specified type?

Also the assertion id seems to indicate property (mentioning observeproperty) but text says events. They are sometimes similar but the semantics behind are different.

File: index.html [hunk 2]
```diff
@@ -2968,12 +2969,305 @@ <h2>Protocol Binding</h2>
...
+  
+          <section class="ednote"> This is inconsistent: there are currently two different unobserve mechanisms
+            - subscriptionId in URL or in payload
+          </section>
+  
+          <p>
+            <span class="rfc2119-assertion" id="http-webhook-profile-protocol-binding-unobserveproperty-1">
+              In order to stop observing a property, a Consumer MUST terminate
+              the corresponding subscription with the Web Thing using the
+              <code>unobserve</code> operation of the <code>Property</code> resource.</span>
```
Comment (egekorkan) [inline index.html:3095]:
```suggestion
              <code>unobserveproperty</code> operation of the <code>Property</code> resource.</span>
```

**Review by egekorkan — CHANGES_REQUESTED** [review]
Mostly minor operation names

File: index.html (hunk 1 above)
Comment (mlagally) [reply index.html:3077]: Thanks for noticing - That's a copy paste error - fixed

**Comment by benfrancis** [comment]
Many of the points from my review of the events protocol binding (#375) also apply to this proposed properties binding, since the assertions have been copied across from there. I would suggest resolving those points before landing this and duplicating all of the same issues.

In particular, I would recommend:
- [ ] Change `observeproperty` and `observeallproperties` responses to `201` with a subscription URI in a header rather than an ID in the body. Add a `Content-type` header and remove the `Accept` header on `observeproperty` and `observeallproperties` requests to reflect this.
- [ ] Change `unobserveproperty` and `unobserveallproperties` requests to a `DELETE` on the subscription URL, with a `204` response and an example

I think these changes are important so that the protocol binding doesn't rely on a fixed URL structure (which we have so far avoided in all of the other protocol bindings). This would also resolve the current issue with two different unsubscribe mechanisms, one of which has strange semantics (`POST` to `/properties/level` is subscribe and `POST` to `properties/level/123-456-789` is unsubscribe), and the other is ambiguous (both operations are `POST` requests on the same URL, so presumably would have to be differentiated by the content in the body though that isn't shown). From a REST point of view it makes more sense to me to create a subscription resource with the ID in its URI to represent a subscription, then delete that resource to cancel the subscription. This would be consistent with what we do in the actions protocol binding for cancelling something.

The subscription URI could alternatively be included in both a header _and_ the body of the `observeproperty` response (which was the compromise we came up with for the actions protocol binding), but it should definitely be a URI and not just an ID.

Much of my feedback on the Event Connections section in #375 also applies here:
- [ ] Remove the assertion "If the connection between the Web Thing and the Consumer drops, the Web Thing MUST re-establish the connection."
- [ ] Remove the assertion "Once the connection is re-established the Web Thing SHOULD, if possible, send any missed notifications which occurred since the last successful notification." 

(These assertions have been copied and pasted from the SSE protocol binding, but they don't make sense for Webhooks where every property reading would be sent in a separate HTTP request, and therefore a separate TCP "connection".)

- [ ] Add assertions to say that property readings must be sent from the Thing to the Consumer with an HTTP `POST` request with a `Content-Type` header set to `application/json`, with a `200` or `204` response.

My suggestions about batched event payloads, event rate limiting, mutual authentication and abuse protection also apply here, but can be addressed separately.

File: index.html [hunk 3]
```diff
@@ -3330,41 +3330,34 @@ <h4><code>subscribeevent</code></h4>
...
             <li>
-              <code>Accept</code> header set to <code>application/json</code>
+              <code>Location</code> header set to the URL of the <code>Event</code> resource
+              followed by a unique <code>subscriptionId</code>
             </li>
           </ul>
           <pre class="example">
-              HTTP/1.1 200 OK
+              HTTP/1.1 201 Created
               Content-Type: application/json
```
Comment (lu-zero) [inline index.html:3345]: is the Content-Type still needed?

**Comment by mlagally** [comment]
@benfrancis 
thanks for your thorough review. I addressed most points, also see corresponding PRs #390 #391 and #392

File: index.html (hunk 3 above)
Comment (mlagally) [reply index.html:3355]:
```suggestion
```

**Comment by mlagally** [comment]
Approve to merge in Profile call on April 19th

**Pull request merged**